aspnet/security/authentication/cookieauthentication.rst

```diff
@@ -0,0 +1,116 @@
+
+ .. code-block:: c#
+  
+  await HttpContext.Authentication.SignInAsync("MyCookieMiddlewareInstrance", principal);
+
+This will create an encrypted cookie and add it to the current response. The ``AuthenticationScheme`` specified during :ref:`configuration <security-authentication-cookie-middleware-configuring>` must also be used when calling ``SignInAsync``.
+
+Under the covers the encryption used is ASP.NET's :ref:`Data Protection <security-data-protection-getting-started>` system. If you are hosting on multiple machines, load balancing or using a web farm then you will need to :ref:`configure <data-protection-configuring>` data protection to use the same key ring and application identifier.
+
+Signing out
+-----------
+
+To sign out the current user, and delete their cookie call the following inside your controller
+
+.. code-block:: c#
+
+ await HttpContext.Authentication.SignOutAsync("MyCookieMiddlewareInstrance");
+
+Reacting to back-end changes
+----------------------------
+
+.. warning::
+  Once a principal cookie has been created it becomes the single source of identity - even if you disable a user in your back-end systems the cookie middleware has no knowledge of this and a user will continue to stay logged in as long as their cookie is valid. 
+
+The cookie authentication middleware provides a series of Events in its option class. The ``ValidateAsync()`` event can be used to intercept and override validation of the cookie identity.
+
+Consider a back-end user database that may have a LastChanged column. In order to invalidate a cookie when the database changes you should first, when :ref:`creating the cookie <security-authentication-cookie-middleware-creating-a-cookie>`, add a LastChanged claim containing the current value. Then, when the database changes the LastChanged value should also be updated. 
+
+To implement an override for the ``ValidateAsync()`` event you must write a method with the following signature;
+
+.. code-block:: c#
+
+ Task ValidateAsync(CookieValidatePrincipalContext context);
+
+ASP.NET Identity implements this check as part of its `SecurityStampValidator <https://github.com/aspnet/Identity/blob/dev/src/Microsoft.AspNet.Identity/SecurityStampValidator.cs>`_. An simple example would look something like as follows;
+
+.. code-block:: c#
+
+ public static class LastChangedValidator
+ {
+     public static async Task ValidateAsync(CookieValidatePrincipalContext context)
+     {
+         // Pull database from registered DI services.
+         var userRepository = context.HttpContext.RequestServices.GetRequiredService<IUserRepository>();
+         var userPrincipal = context.Principal;
+
+         // Look for the last changed claim.
+         string lastChanged;
+         lastChanged = (from c in userPrincipal.Claims
```

Comment:
nit: you could also use `userPrincipal.Claims.FindFirst("LastUpdated")?.Value` (or `userPrincipal.Claims.FindFirstValue("LastUpdated")` when referencing Identity)